Source/TCCProfileImporter/TCCProfileImporter.swift

```diff
@@ -0,0 +1,124 @@
+//  copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+//  SOFTWARE.
+//
+
+import Foundation
+import AppKit
+
+enum TCCProfileImportResult {
+    case success(TCCProfile)
+    case failure(TCCProfileImportError?)
+}
+
+typealias TCCProfileImportCompletion = ((TCCProfileImportResult) -> Void)
+
+
+/// Load tcc profiles
+public class TCCProfileImporter {
+
+    // MARK: Load TCCProfile
+
+    /// Load TCC Profile data from file
+    ///
+    /// - Parameter completion: TCCProfileImportCompletion - success with TCCProfile or failure with TCCProfileImport Error
+    func loadTCCProfileFromFile(window: NSWindow, _ completion: @escaping TCCProfileImportCompletion) {
+        let openPanel = NSOpenPanel.init()
+        openPanel.allowedFileTypes = ["mobileconfig", "plist"]
+        openPanel.allowsMultipleSelection = false
+        openPanel.canChooseDirectories = false
+        openPanel.canCreateDirectories = false
+        openPanel.canChooseFiles = true
+        openPanel.title = "Open TCCProfile File"
+
+        openPanel.beginSheetModal(for: window) { (response) in
+            if response != .OK {
+                // Cancelled
+                completion(TCCProfileImportResult.failure(nil))
+            } else {
+                if let result = openPanel.url {
+                    self.decodeTCCProfile(fileUrl: result, { tccProfileResult in
+                        return completion(tccProfileResult)
+                    })
+                } else {
+                    completion(TCCProfileImportResult.failure(TCCProfileImportError.unableToOpenFile))
+                }
+            }
+        }
+
+    }
+
+    /// Mapping & Decoding tcc profile
+    ///
+    /// - Parameter fileUrl: path with a file to load, completion: TCCProfileImportCompletion - success with TCCProfile or failure with TCCProfileImport Error
+    func decodeTCCProfile(fileUrl: URL, _ completion: @escaping TCCProfileImportCompletion) {
+        let contents: Data
+        do {
+            contents = try Data(contentsOf: fileUrl)
+        } catch {
+            return completion(TCCProfileImportResult.failure(TCCProfileImportError.unableToOpenFile))
+        }
+
+        var unsignedResult = self.unsignTCCProfile(fileData: contents)
+
+        if unsignedResult == nil {
+            unsignedResult = contents
+        }
+
+        guard let tccProfileData = unsignedResult else {
+            return completion(TCCProfileImportResult.failure(TCCProfileImportError.decodeProfileError))
+        }
```

Comment:
It is confusing. I tried simplifying by bringing in SwiftyProvisioningProfile with SPM but it parses into a model that is more generic ([SwiftyProvisioningProfile](https://github.com/Sherlouk/SwiftProvisioningProfile/blob/master/Sources/SwiftyProvisioningProfile/Model/ProvisioningProfile.swift))
I tried to follow the code borrowed from SwiftProvisioningProfile package more closely.
Long term it might be better to update the SwiftProvisioningProfile package to include a TCCProfile model with its own parse method.